### PR TITLE
#758: applied_taxes of empty cart causes 'Internal server error' message on PHP 7.2

### PR DESCRIPTION
**Pull request by therealtbs**
### Description (*)

From PHP 7.2 onward there is a warning generated when attempting to call `count()` on anything that is not an array or an object implementing Countable, when a cart is created, initially the appliedTaxes are null until a product is added.

### Fixed Issues (if relevant)

1. magento/graphql-ce#758: Cart applied_taxes of empty cart causes 'Internal server error' message on PHP 7.2

### Manual testing scenarios (*)

1. See reproduction steps of magento/graphql-ce#758

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lenaorobei** [comment]
@therealtbs thank you for the contribution!
Could you please cover this case with `api-functional` test?

_Examples:_
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Customer
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer

_Guide:_
https://devdocs.magento.com/guides/v2.3/graphql/functional-testing.html

**Comment by naydav** [comment]
Hi @therealtbs,

please cover this case (try to load cart taxes if taxes are not set) with API-functional tests
thanks

Examples:
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Customer
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer

Guide:
https://devdocs.magento.com/guides/v2.3/graphql/functional-testing.html

File: app/code/Magento/QuoteGraphQl/Model/Resolver/CartPrices.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ private function getAppliedTaxes(Total $total, string $currency): array
         $appliedTaxesData = [];
         $appliedTaxes = $total->getAppliedTaxes();
 
-        if (count($appliedTaxes) === 0) {
+        if ($appliedTaxes === null || count($appliedTaxes) === 0) {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/Model/Resolver/CartPrices.php]:
``` suggestion
        if (empty($appliedTaxes)) {
```

Comment (therealtbs) [reply app/code/Magento/QuoteGraphQl/Model/Resolver/CartPrices.php]: I assume you meant `empty` instead of `!empty`?

Comment (lenaorobei) [reply app/code/Magento/QuoteGraphQl/Model/Resolver/CartPrices.php]: Sure, you're right! 👍

**Comment by naydav** [comment]
Hi @TomashKhamlai 
Please verify the fix.
Thanks

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-5436](https://jira.corp.magento.com/browse/ENGCOM-5436) has been created to process this Pull Request
:eight_spoked_asterisk: @VladimirZaets, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@therealtbs thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by magento-engcom-team** [comment]
@therealtbs thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by magento-engcom-team** [comment]
@therealtbs thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Pull request merged**